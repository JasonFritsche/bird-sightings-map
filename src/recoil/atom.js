import { atom, selector } from 'recoil'

const allMarkers = atom({
  key: 'allMarkers',
  default: []
})

const selectedBounds = atom({
  key: 'selectedBounds',
  default: { _northEast: { lat: null, lng: null }, _southWest: { lat: null, lng: null } }
})

const visibleFilters = selector({
  key: 'visibleFilters',
  get: ({ get }) => {
    const markers = get(allMarkers)
    return markers.filter((m) => m.isVisible)
  }
})

export { allMarkers, visibleFilters, selectedBounds }
