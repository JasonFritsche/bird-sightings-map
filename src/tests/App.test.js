import React from 'react'
import { create } from 'react-test-renderer'
import App from '../App'
import { RecoilRoot } from 'recoil'

describe('App component', () => {
  test('App component snapshot', () => {
    const app = create(
      <RecoilRoot>
        <App />
      </RecoilRoot>
    )
    expect(app.toJSON()).toMatchSnapshot()
  })
})
