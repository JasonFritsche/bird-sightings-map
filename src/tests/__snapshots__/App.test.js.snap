// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App component snapshot 1`] = `
<div
  className="w-full h-screen"
>
  <div
    className="drawer drawer-mobile rounded-lg shadow bg-base-200 h-full"
  >
    <input
      className="drawer-toggle"
      id="the-drawer"
      type="checkbox"
    />
    <div
      className="flex flex-col items-center justify-center drawer-content"
    >
      <div />
    </div>
    <div
      className="drawer-side"
    >
      <label
        className="drawer-overlay"
        htmlFor="the-drawer"
      />
      <ul
        className="menu p-4 overflow-y-auto w-80 bg-base-100 text-base-content"
      >
        <li>
          <div>
            <h1>
              Bird Sightings
            </h1>
          </div>
        </li>
        <li>
          <div
            className="divider"
          >
            Search
          </div>
        </li>
        <li>
          <div>
            <button
              className="btn btn-wide btn-sm"
              onClick={[Function]}
            >
              Search
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
